mobius/src/geometry/line_segment.rs

```diff
@@ -12,6 +12,18 @@ impl LineSegment {
     pub fn new(start: Complex, end: Complex) -> Self {
         Self { start, end }
     }
+
+    pub fn reverse(&self) -> Self {
```

Comment:
This should be part of the `DirectedEdge` trait

Reply:
Consider also a testing macro that adds the following test:

- reversing edge (a, b) gives edge (b, a) (i.e. start() and end() swap)

Reply:
Not every directed edge is reversible (e.g. a ray has no reverse. A line segment between a point and infinity would, but that's not how this is expressed).

Also, adding this to the trait would make it not object-safe. 

So nevermind about this idea!